torch/_prims/__init__.py

```diff
@@ -2323,10 +2323,12 @@ def _arange_meta(
         step != 0,
         lambda: "step must be nonzero",
     )
-    utils.check(
-        math.isfinite(start) and math.isfinite(end),
-        lambda: f"unsupported range: {start} -> {end}",
-    )
+    # SymInts can't represent inf
+    if not isinstance(start, torch.SymInt) and not isinstance(end, torch.SymInt):
+        utils.check(
+            math.isfinite(start) and math.isfinite(end),
```

Comment:
how to add sym_isfinite?

Reply:
`sym_isfinite` is pretty easy for int; we just test if is SymInt and return True unconditionally. For float need to actually translate into some sort of condition, but probably doesn't matter too much.